src/Coordination/ACLMap.cpp

```diff
@@ -42,11 +42,14 @@ bool ACLMap::ACLsComparator::operator()(const Coordination::ACLs & left, const C
 
 uint64_t ACLMap::convertACLs(const Coordination::ACLs & acls)
 {
+    if (acls.empty())
+        return 0;
+
     if (acl_to_num.count(acls))
         return acl_to_num[acls];
 
     /// Start from one
-    auto index = acl_to_num.size() + 1;
+    auto index = max_acl_id++;
```

Comment:
Can you describe problem in details? As far as I can see we will restore acl_to_num and num_to_acl from logs/snapshot and everything should work well. Why it will erase?

Reply:
It will erase from ```removeUsage``` method.

Reply:
But why it's a problem? If nobody uses some id from map than we can reuse it again?

Reply:
Because after erasing, the size will be reduced. When a new acls comes in, it will be assigned an index equal to the current maximum index so that the original acls will be overwritten.

Reply:
But and `ID` can be erased from map only when nobody use this `ID` and should be safe to reuse it?

Reply:
I add some tests to be more intuitive.